src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,636 @@
+        
+        String jsonResults = buildJsonOutput();
+        LOG.debug("JSON results: " + jsonResults);
+
+        List<AmazonAccountDetail> amazonAccountDetails = buildAmazonAcountListFromJson(jsonResults);
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService() Number of amazon account details received: " + amazonAccountDetails.size());
+        
+        boolean valid = true;
+        
+        if (amazonAccountDetails.size() == 0) {
+            LOG.error("generateDistributionOfIncomeDocumentsFromAWSService() No amazon account detail objects parsed from the webservice.");
+            valid =  false;
+        }
+        
+        valid = valid && processAmazonAccountDetails(amazonAccountDetails);
+        
+        clearMemberVariables();
+        if (!valid) {
+            throw new RuntimeException("There was error processing AWS Billing.");
+        }
+    }
+    
+    private String buildJsonOutput() {
+        ClientConfig clientConfig = new DefaultClientConfig();
+        Client client = Client.create(clientConfig);
+        
+        ClientResponse response = client.handle(buildClientRequest());
+        
+        response.bufferEntity();
+        String results = response.getEntity(String.class);
+        return results;
+    }
+    
+    protected ClientRequest buildClientRequest() {
+        ClientRequest.Builder builder = new ClientRequest.Builder();
+        builder.accept(MediaType.APPLICATION_JSON);
+        builder.header(CuFPConstants.AmazonWebServiceBillingConstants.AUTHORIZATION_HEADER_NAME, 
+                CuFPConstants.AmazonWebServiceBillingConstants.AUTHORIZATION_TOKEN_VALUE_STARTER + getAwsToken());
+        URI uri;
+        try {
+            uri = new URI(getAwsURL() + CuFPConstants.AmazonWebServiceBillingConstants.URL_PARAMETER_YEAR + findProcessYear() + 
+                    CuFPConstants.AmazonWebServiceBillingConstants.URL_PARAMETER_MONTH + findProcessMonthNumber());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        ClientRequest request = builder.build(uri, CuFPConstants.AmazonWebServiceBillingConstants.HTTP_METHOD_GET_NAME);
+        return request;
+    }
+    
+    protected List<AmazonAccountDetail> buildAmazonAcountListFromJson(String jsonResults) {
+        ObjectMapper mapper = new ObjectMapper();
+        AmazonAccountDetailContainer detailContainer = new AmazonAccountDetailContainer();
+        try {
+            detailContainer = mapper.readValue(jsonResults, AmazonAccountDetailContainer.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return detailContainer.getAccountDetail();
+    }
+
+    protected boolean processAmazonAccountDetails(List<AmazonAccountDetail> amazonAccountDetails) {
+        boolean valid = true;
```

Comment:
Since this method is not doing validation could there be a better name for this variable?